packages/amplify-graphql-relational-transformer/src/utils.ts

```diff
@@ -243,8 +252,9 @@ const getReferencesAssociatedField = (
         .includes(directive.name.value);
     });
 
+    // The field has the right base type, but isn't the proper association, so skip to the next candidate
     if (!associatedRelationalDirective) {
-      throw new InvalidDirectiveError(`Uni-directional relationships are not supported. ${expectedBidirectionalErrorMessages()}`);
+      return [];
```

Comment:
This is safe to do because we have validation elsewhere for empty associated fields. This check was causing an early return in recursive schemas.